### DISCUSSION TITLE
How to calculate a group of shapes' wrapping rect including shadows for exporting to PNG?

### DISCUSSION THREAD

**Discussion by lanyusan** [konvajs/konva#1261] [Q&A] [answered]
Hi, 

We have built a functioning prototype with Konva and it looks very promising. We are now working on exporting function.

The use case is that we will have a group of shapes that need to be exported exactly the size of the wrapping rect that covers all the shapes, including shadows of all the shapes.

How to calculate this wrapping rect? 

I understand there is a `getClientRect` function but it is for on screen position. It needs to be translated to stage coordinates including scaling and stage x, y. Is there a function to do the translation?

Or is there a better way to get the exact `x, y, width, height`  of the wrapping rect?

Thanks again for this great lib.

**Comment by lavrton** [comment, marked as answer]
I don't really understand what you can't use `node.getClientRect()`. It should give you exactly what you want. If you need relative results, you can use `relativeTo` argument:

```js
// calculate clean rectangle relative to stage node
// it means it will ignore stage transforms
node.getClientRect({ relativeTo: stage });
```

**Reply by lanyusan** [reply]
Thanks for the reply. Didn't know this trick `{ relativeTo: stage }`
